scripts/xcode_cloud_helper.rb

```diff
@@ -0,0 +1,763 @@
+    puts "Token -> #{token}"
+end
+
+def get_workflows
+    product_id = get_realm_product_id
+    response = get("/ciProducts/#{product_id}/workflows?limit=200")
+    result = JSON.parse(response.body)
+    list_workflows = []
+    result.collect do |doc|
+        doc[1].each { |workflow|
+            if workflow.class == Hash
+                list_workflows.append(workflow)
+            end
+        }
+    end
+    return list_workflows
+end
+
+def get_products
+    response = get("/ciProducts")
+    result = JSON.parse(response.body)
+    list_products = []
+    result.collect do |doc|
+        doc[1].each { |prod|
+            if prod.class == Hash
+                list_products.append({ "product" => prod["attributes"]["name"], "id" => prod["id"], "type" => prod["attributes"]["productType"] })
+            end
+        }
+    end
+    return list_products
+end
+
+def get_repositories
+    response = get("/scmRepositories")
+    result = JSON.parse(response.body)
+    list_repositories = []
+    result.collect do |doc|
+        doc[1].each { |repo|
+            if repo.class == Hash
+                list_repositories.append({ "name" => repo["attributes"]["repositoryName"], "id" => repo["id"], "url" => repo["attributes"]["httpCloneUrl"] })
+            end
+        }
+    end
+    return list_repositories
+end
+
+def get_macos_versions
+    response = get("/ciMacOsVersions")
+    result = JSON.parse(response.body)
+    list_macosversion = []
+    result.collect do |doc|
+        doc[1].each { |macos|
+            if macos.class == Hash
+                list_macosversion.append({ "name" => macos["attributes"]["name"], "id" => macos["id"], "version" => macos["attributes"]["version"] })
+            end
+        }
+    end
+    return list_macosversion
+end
+
+def get_xcode_versions
+    response = get("/ciXcodeVersions")
+    result = JSON.parse(response.body)
+    list_xcodeversion = []
+    result.collect do |doc|
+        doc[1].each { |xcode|
+            if xcode.class == Hash
+                list_xcodeversion.append({ "name" =>  xcode["attributes"]["name"], "id" => xcode["id"], "version" => xcode["attributes"]["version"] })
+            end
+        }
+    end
+    return list_xcodeversion
+end
+
+def print_workflow_info(id)
+    workflow_info = get_workflow_info(id)
+    puts "Workflow Info:"
+    puts workflow_info
+end
+
+def get_workflow_info(id)
+    response = get("/ciWorkflows/#{id}")
+    return response.body
+end
+
+def get(path)
+    url = "#{APP_STORE_URL}#{path}"
+    uri = URI.parse(url)
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = true
+    request = Net::HTTP::Get.new(uri.request_uri)
+    request["Authorization"] = "Bearer #{JWT_BEARER}"
+    request["Accept"] = "application/json"
+    response = http.request(request)
+
+    if response.code == "200"
+        return response
+    else
+        puts "ERROR!!! #{response.code} #{response.body}"
```

Comment:
Probably want to throw an error in each of these spots rather than just printing and then hitting a nil reference error later.